src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java

```diff
@@ -0,0 +1,347 @@
+ * for accessing services that use OAuth for authorization.
+ * <p>
+ * Supports most common signature inclusion and calculation methods: HMAC-SHA1 for
+ * calculation, and Header inclusion as inclusion method. Nonce generation uses
+ * simple random numbers with base64 encoding.
+ *
+ * @author tatu (tatu.saloranta@iki.fi)
+ */
+public class OAuthSignatureCalculatorRSA implements SignatureCalculator {
+    public final static String HEADER_AUTHORIZATION = "Authorization";
+
+    private static final String KEY_OAUTH_CONSUMER_KEY = "oauth_consumer_key";
+    private static final String KEY_OAUTH_NONCE = "oauth_nonce";
+    private static final String KEY_OAUTH_SIGNATURE = "oauth_signature";
+    private static final String KEY_OAUTH_SIGNATURE_METHOD = "oauth_signature_method";
+    private static final String KEY_OAUTH_TIMESTAMP = "oauth_timestamp";
+    private static final String KEY_OAUTH_TOKEN = "oauth_token";
+    private static final String KEY_OAUTH_VERSION = "oauth_version";
+
+    private static final String OAUTH_VERSION_1_0 = "1.0";
+    private static final String OAUTH_SIGNATURE_METHOD = "RSA-SHA1";
+
+    protected static final ThreadLocal<byte[]> NONCE_BUFFER = new ThreadLocal<byte[]>() {
+        protected byte[] initialValue() {
+            return new byte[16];
+        }
+    };
+
+    protected final ConsumerKey consumerAuth;
+
+    protected final RequestToken userAuth;
+
+    /**
+     * @param consumerAuth Consumer key to use for signature calculation
+     * @param userAuth     Request/access token to use for signature calculation
+     */
+    public OAuthSignatureCalculatorRSA(ConsumerKey consumerAuth, RequestToken userAuth) {
+        this.consumerAuth = consumerAuth;
+        this.userAuth = userAuth;
+    }
+
+    @Override
+    public void calculateAndAddSignature(Request request, RequestBuilderBase<?> requestBuilder) {
+        String nonce = generateNonce();
+        long timestamp = generateTimestamp();
+        String signature = null;
+        try {
+            signature = calculateSignature(request.getMethod(), request.getUri(), timestamp, nonce, request.getFormParams(), request.getQueryParams());
+        } catch (OAuthMessageSignerException e) {
+            e.printStackTrace();
+        }
+        String headerValue = constructAuthHeader(signature, nonce, timestamp);
+        requestBuilder.setHeader(HEADER_AUTHORIZATION, headerValue);
+    }
+
+    private String baseUrl(Uri uri) {
+        /* 07-Oct-2010, tatu: URL may contain default port number; if so, need to extract
+         *   from base URL.
+         */
+        String scheme = uri.getScheme();
+
+        StringBuilder sb = StringUtils.stringBuilder();
+        sb.append(scheme).append("://").append(uri.getHost());
+
+        int port = uri.getPort();
+        if (scheme.equals("http")) {
+            if (port == 80)
+                port = -1;
+        } else if (scheme.equals("https")) {
+            if (port == 443)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [These nested if statements could be combined](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903587/issues/source?bid=2734487&fileBranchId=3159563#l115)